corehq/apps/userreports/es/adapter.py

```diff
@@ -207,6 +207,14 @@ def _save_rows(self, rows, doc):
                 id=normalize_id(primary_key_values), doc_type="indicator"
             )
 
+    def doc_exists(self, doc):
+        records = self.get_query_object().es.doc_id(doc['_id'])
+        return records.count > 0
+
+    def delete(self, doc):
+        if self.doc_exists(doc):
```

Comment:
This means that doc_exists will be called twice for any deletion. I think instead of checking if it exists again, there should be a try/except block. otherwise you could keep the doc_exists here and just call `delete` in an `else` statement in the pillow.